### PR TITLE
add `delete-kargs-hardening` and documentation

### PR DESCRIPTION
**Pull request by czhang03**
Groups all kargs into two variables, and added additional documentation. Replaces https://github.com/secureblue/secureblue/pull/199

### CODE REVIEW

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
     echo "Warning: setting these kargs may lead to boot issues on some hardware."
-    rpm-ostree kargs \
-      --append-if-missing="efi=disable_early_pci_dma"
+    rpm-ostree kargs --append-if-missing="{{harden_karg_unstable}}"
+
+# delete boot parameters for hardening
+delete-kargs-hardening:
+    rpm-ostree kargs --delete-if-present="{{harden_karg}} {{harden_karg_unstable}}"
 
+# enable hardenmalloc for flatpak
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: please remove these comments, the only information they provide is already captured in the function names

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 2]
```diff
@@ -1,36 +1,47 @@
...
 
 set-kargs-hardening-unstable:
     echo "Warning: setting these kargs may lead to boot issues on some hardware."
-    rpm-ostree kargs \
-      --append-if-missing="efi=disable_early_pci_dma"
+    rpm-ostree kargs --append-if-missing="{{harden_karg_unstable}}"
+
+# delete boot parameters for hardening
+delete-kargs-hardening:
+    rpm-ostree kargs --delete-if-present="{{harden_karg}} {{harden_karg_unstable}}"
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: These variable names don't match the variable names above. This indicates you didn't test this functionality. Please provide a screenshot showing test results.

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 3]
```diff
@@ -1,36 +1,47 @@
 import '100-bling.just'
 # Include some of your custom scripts here!
 
+# hardening Kernel Arguments
+harden_kargs := "init_on_alloc=1" + \
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: Does this work? It's missing spaces in-between the strings.

**Review by RoyalOughtness — CHANGES_REQUESTED** [review]

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 2 above)
Comment (czhang03) [reply config/files/usr/share/ublue-os/just/60-custom.just]: Sorry, I discovered the variable naming problem right before my commit. I have tested using `just --show` before the variable name change.

I will fix this when I am around my computer.

Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just]: `just --show` doesn't test anything. You need to run all of the commands and confirm their functionality.

Comment (czhang03) [reply config/files/usr/share/ublue-os/just/60-custom.just]: Yes, I understand. I was in a rush and didn't have time to wait until the karg applies.

Thanks for catching it, and sorry for the mistake.

Comment (RoyalOughtness) [reply config/files/usr/share/ublue-os/just/60-custom.just]: > didn't have time to wait until the karg applies.

Please re-review the CONTRIBUTING guide. https://github.com/secureblue/secureblue/blob/live/CONTRIBUTING.md#pull-requests

Per the guide, please do not open pull requests until your changes are fully tested.

**Comment by czhang03** [comment]
I have been testing this change, it seems like monolithic string confuses `rpm-ostree karg`. 

If the karg is present, using a long string will cause duplicate:
```
cheng@cheng-fedora-laptop:~$ just set-kargs-hardening
rpm-ostree kargs --append-if-missing="init_on_alloc=1 init_on_free=1 slab_nomerge page_alloc.shuffle=1 randomize_kstack_offset=on vsyscall=none debugfs=off lockdown=confidentiality random.trust_cpu=off random.trust_bootloader=off iommu=force intel_iommu=on amd_iommu=force_isolation iommu.passthrough=0 iommu.strict=1 pti=on module.sig_enforce=1 mitigations=auto,nosmt" 
Checking out tree ...
Changes queued for next boot. Run "systemctl reboot" to start a reboot

$ rpm-ostree kargs 
... init_on_alloc=1 init_on_free=1 slab_nomerge page_alloc.shuffle=1 randomize_kstack_offset=on vsyscall=none lockdown=confidentiality random.trust_cpu=off random.trust_bootloader=off iommu=force intel_iommu=on amd_iommu=force_isolation iommu.passthrough=0 iommu.strict=1 pti=on module.sig_enforce=1 mitigations=auto,nosmt amdgpu.sg_display=0 rtc_cmos.use_acpi_alarm=1 debugfs=off init_on_alloc=1 init_on_free=1 slab_nomerge page_alloc.shuffle=1 randomize_kstack_offset=on vsyscall=none debugfs=off lockdown=confidentiality random.trust_cpu=off random.trust_bootloader=off iommu=force intel_iommu=on amd_iommu=force_isolation iommu.passthrough=0 iommu.strict=1 pti=on module.sig_enforce=1 mitigations=auto,nosmt
```
`...` represents unrelated information.

similarly, deleting karg will not be successful:
```
$ just delete-kargs-hardening
rpm-ostree kargs --delete-if-present="init_on_alloc=1 init_on_free=1 slab_nomerge page_alloc.shuffle=1 randomize_kstack_offset=on vsyscall=none debugfs=off lockdown=confidentiality random.trust_cpu=off random.trust_bootloader=off iommu=force intel_iommu=on amd_iommu=force_isolation iommu.passthrough=0 iommu.strict=1 pti=on module.sig_enforce=1 mitigations=auto,nosmt efi=disable_early_pci_dma"
No changes.
```

My guess is `rpm-ostree` simply try to parse the input on the first `=`, which leads to the key being: `init_on_alloc` and value being `1 init_on_free=1 slab_nomerge page_alloc.shuffle=1 ...`

This also matches the syntax in help page of `rpm-ostree kargs`:
```
  --append-if-missing=KEY=VALUE     
  --delete-if-present=KEY=VALUE     
```

I will try to see if there is better way to do this.

**Comment by RoyalOughtness** [comment]
> I will try to see if there is better way to do this.

Thanks, it might be an upstream bug we need to create to `rpm-ostree` itself

**Comment by czhang03** [comment]
This is giving more trouble than it is worth. I don't think just file support variable list. 

Also, if we add or remove variables in the future, the delete command might not even remove all the hardened kargs. For example, if people run `set-kargs-harden` when `init_on_alloc=1` is in the harden parameters, and try to remove them when `init_on_alloc=1` is no longer part of our harden parameters, then `init_on_alloc=1` will not be removed.

I think it is better for users to try to delete them one-by-one at this point.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 2 above)
Comment (czhang03) [reply config/files/usr/share/ublue-os/just/60-custom.just]: I understand, I will fully test the changes next time. Thanks for catching the mistake.

**Comment by RoyalOughtness** [comment]
> This is giving more trouble than it is worth. I don't think just file support variable list.
> 
> Also, if we add or remove variables in the future, the delete command might not even remove all the hardened kargs. For example, if people run `set-kargs-harden` when `init_on_alloc=1` is in the harden parameters, and try to remove them when `init_on_alloc=1` is no longer part of our harden parameters, then `init_on_alloc=1` will not be removed.
> 
> I think it is better for users to try to delete them one-by-one at this point.

That's fair, thanks for your efforts and for looking into it

**Pull request closed without merging**